src/ipc.rs

```diff
@@ -62,12 +62,12 @@ pub fn bytes_channel() -> Result<(IpcBytesSender, IpcBytesReceiver),Error> {
 }
 
 #[derive(Debug)]
-pub struct IpcReceiver<T> where T: Deserialize + Serialize {
+pub struct IpcReceiver<T> where T: for<'de> Deserialize<'de> + Serialize {
```

Comment:
`Deserialize<'static>` is never what you want. `for<'de> Deserialize<'de>` or `DeserializeOwned` is correct. See https://serde.rs/lifetimes.html#trait-bounds.